container/writer.go

```diff
@@ -103,6 +103,11 @@ func (avroWriter *Writer) Flush() error {
 		fwWriter.Reset(avroWriter.blockBuffer)
 	}
 
+	// dont flush if unused
+	if !avroWriter.headerWritten {
+		return nil
+	}
+
```

Comment:
Farther down we have 

avroWriter.blockBuffer.Reset()
avroWriter.nextBlockRecords = 0

I don't know if they are necessary.

Perhaps we just wrap the serialization in
``` if avroWriter.headerWritten {
}
``` ?